src/packages/utils/src/things/json-rpc/json-rpc-data.ts

```diff
@@ -1,57 +1,95 @@
 import { BaseJsonRpcType } from "./json-rpc-base-types";
-import { strCache, toStrings } from "./json-rpc-base-types";
+import { BUFFER_EMPTY } from "../../utils/constants";
 
-function validateByteLength(byteLength?: number) {
-  if (typeof byteLength !== "number" || !(byteLength >= 0)) {
-    throw new Error(`byteLength must be a number greater than or equal to 0, provided: ${byteLength}`);
-  }
-}
+export type JsonRpcDataInputArg = string | Buffer;
 
 export class Data extends BaseJsonRpcType {
+  public static Empty = Data.from(BUFFER_EMPTY);
 
-  constructor(value: string | Buffer, private _byteLength?: number) {
+  constructor(value: JsonRpcDataInputArg, private _byteLength?: number) {
     super(value);
+
     if (typeof value === "bigint") {
       throw new Error(`Cannot create a ${typeof value} as a Data`);
     }
-    if (_byteLength !== undefined) {
-      validateByteLength(_byteLength);
+  }
+
+  public toString(byteLength?: number): string | null {
+    const length = byteLength || this._byteLength;
+
+    if (this.bufferValue == null) {
+      return "0x";
+    }
+
+    if (length === undefined) {
+      return super.toString();
     }
+
+    const strValue = this.bufferValue.toString("hex");
+    return length === undefined? `0x${strValue}` : `0x${Data.stringToFixedByteLength(strValue, length)}`;
   }
-  public toString(byteLength?: number): string {
-    if (byteLength === undefined) {
-      byteLength = this._byteLength;
+
+  public toBuffer(byteLength?: number): Buffer {
+    const length = byteLength || this._byteLength;
+
+    if (this.bufferValue == null) {
+      return BUFFER_EMPTY;
+    }
```

Comment:
The following is probably faster in case where `this.bufferValue` is null/undefined, and the same speed when it is not.

```suggestion
    if (this.bufferValue == null) {
      return BUFFER_EMPTY;
    }
    
    const length = byteLength || this._byteLength;
```